### PR TITLE
ESPnet-Codec decoding and Scoring

### PR DESCRIPTION
**Pull request by ftshijt**
## What?

Codec Training decoding and scoring

(Together with some fixes for the previous PR)

## TODOs in the next one
- [ ] More scoring setup (several optional metrics to be added, such as the existing ones supported in TTS eval)
- [ ] Other metrics for codec (Bitrate, etc.)
- [ ] Model packing and uploading

### CODE REVIEW

**Comment by ftshijt** [comment]
@jctian98 @wyh2000 Please help review this PR as it contains a few updates related to your new PRs. Thx!

File: espnet2/gan_codec/shared/loss/freq_loss.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        normalized (bool): Whether to use normalized one.
+        onesided (bool): Whether to use oneseded one.
+        log_base (Optional[float]): Log base value.
+        alphas (bool): Whether to use alphas as coefficients or not..
+    """
+
+    def __init__(
+        self,
+        fs: int = 22050,
+        range_start: int = 6,
```
Comment (wyh2000) [inline espnet2/gan_codec/shared/loss/freq_loss.py:36]: This function has a conflict with configuration in `train_soundstream2.yaml`. I think train_soundstream_debug.yaml has the correct configuration.

Comment (ftshijt) [reply espnet2/gan_codec/shared/loss/freq_loss.py:36]: thx for catching up! Let me clean up the config to match the new design.

**Comment by ftshijt** [comment]
Thanks for the check! I will merge it given its dependency to many following works.

**Pull request merged**